contracts/lib/OnchainNonMerklizedIdentityBase.sol

```diff
@@ -0,0 +1,245 @@
+    }
+
+    /**
+     * @dev IssuanceProof information about the issuance proof
+     */
+    struct IssuanceProof {
+        string _type;
+        uint256[8] coreClaim;
+        IssuerData issuerData;
+        SmtLib.Proof mtp;
+    }
+
+    /**
+     * @dev CredentialData information about the credential.
+     * Can be transformed to W3C verifiable credential
+     */
+    struct CredentialData {
+        string id;
+        string[] context;
+        string[] _type;
+        uint64 expirationDate;
+        uint64 issuanceDate;
+        uint256 issuer;
+        SubjectField[] credentialSubject;
+        Status credentialStatus;
+        Schema credentialSchema;
+        IssuanceProof[] proof;
+    }
+
+    /**
+     * @dev Id
+     */
+    struct Id {
+        uint256 id;
+    }
+
+    function initialize(address _stateContractAddr) public virtual override {
+        IdentityBase.initialize(_stateContractAddr);
+    }
+
+    /**
+     * @dev listUserCredentials. Get list of user credentials identifiers
+     * @param _userId user id
+     */
+    function listUserCredentials(uint256 _userId) external virtual returns (Id[] memory);
+
+    /**
+     * @dev getCredential. Get credential by user id and credential id
+     */
+    function getCredential(
+        uint256 _userId,
+        uint256 _credentialId
+    ) external virtual returns (CredentialData memory);
+
+    /**
+     * @dev processOnchainCredentialData. Build credential data from core claim and additional information
+     */
+    function processOnchainCredentialData(
+        CredentialInformation memory credentialInformation,
+        CredentialMetadata memory credentialMetadata,
+        SubjectField[] memory credentialSubject,
+        Claim memory claim
+    ) internal view returns (CredentialData memory) {
+        IssuerState memory issuerState = IssuerState({
+            rootOfRoots: identity.latestPublishedTreeRoots.rootsRoot,
+            claimsTreeRoot: identity.latestPublishedTreeRoots.claimsRoot,
+            revocationTreeRoot: identity.latestPublishedTreeRoots.revocationsRoot,
+            value: identity.latestPublishedState
+        });
+
+        IssuerData memory issuerData = IssuerData({id: identity.id, state: issuerState});
+
+        SmtLib.Proof memory mtp = identity.getClaimProof(claim.hashIndex);
+        require(mtp.existence, "Claim does not exist in issuer state");
+
+        IssuanceProof memory issuanceProof = IssuanceProof({
+            _type: "Iden3SparseMerkleTreeProof",
+            coreClaim: claim.coreClaim,
+            issuerData: issuerData,
+            mtp: mtp
+        });
+        IssuanceProof[] memory proofs = new IssuanceProof[](1);
+        proofs[0] = issuanceProof;
+
+        uint256 chainId;
+        assembly {
+            chainId := chainid()
+        }
+        return
+            CredentialData({
+                id: string(
+                    abi.encodePacked(
+                        "urn:onchain:",
```

Comment:
Probably "urn:onchain:" will not work in json-ld validation. Can we do "did:polygonid:..../credential?id=id" in schema-processor as we do for user id, issuer id and revocation status id?

Reply:
We can't. Since we can't get a string representation of did in the smart contract. I discussed this with Vlad. Everything is working.